source/backend/dal/Repositories/AcquisitionFileRepository.cs

```diff
@@ -124,6 +124,438 @@ public PimsAcquisitionFile GetById(long id)
                 .FirstOrDefault(x => x.AcquisitionFileId == id) ?? throw new KeyNotFoundException();
         }
 
+        /// <summary>
+        /// Retrieves the acquisition file with the specified id last update information.
+        /// </summary>
+        /// <param name="id"></param>
+        /// <returns></returns>
+        public LastUpdatedByModel GetLastUpdateBy(long id)
```

Comment:
The size of this is a bit of a code smell, but I feel like refactoring this to use some kind of generic method might also be messy. Thoughts?

Reply:
I agree. For now I rather leave it as is and create a functor later.